### DISCUSSION TITLE
Fast refresh sometimes does not work

### DISCUSSION THREAD

**Discussion by kota65535** [gatsbyjs/gatsby#37983] [Help] [answered]
Basically, Gatsby’s development server launched by `gatsby develop` shows my code changes within a second after the message "Re-building development bundle".
But sometimes (every about 10 minutes), "Re-building development bundle" does not finish after 10 minutes (I think this is enough time) and then I must restart the development server. This leads to a quite bad development experience.
Unfortunately, I cannot see when and what code changes caused the failure.

Does anyone else have the same problem?

**Comment by zhaohuanyuu** [comment]
which version are you using？

**Reply by kota65535** [reply]
I'm using v5.7.0.
I will try the latest v5.9.0 and check if the problem persists.

**Reply by kota65535** [reply]
Hmm, still have an issue with v5.9.0 (finally I've quitted by Ctrl-C)

![image](https://user-images.githubusercontent.com/10975834/235285313-0e956fa4-c09a-430a-b681-22214d51fbe1.png)

**Reply by kota65535** [reply, marked as answer]
I figured out the problem.
This is because of not Gatsby but the out-of-date [gatsby-plugin-typescript-checker](https://www.gatsbyjs.com/plugins/gatsby-plugin-typescript-checker/). Unfortunately, this plugin is not maintained for 4 years.

Instead of using it, I directly added the latest [fork-ts-checker-webpack-plugin](https://www.npmjs.com/package/fork-ts-checker-webpack-plugin) as a dependency and then modified `gatsby-node.ts` as follows:

```
import { GatsbyNode } from "gatsby"
const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin')

export const onCreateWebpackConfig: GatsbyNode["onCreateWebpackConfig"] = async ({ actions }, options) => {
    plugins: [
      new ForkTsCheckerWebpackPlugin()
    ],
  })
}
```

I think this suggestion should be noted in the [type-checking](https://www.gatsbyjs.com/plugins/gatsby-plugin-typescript/#type-checking) section of gatsby-plugin-typescript documentation.
